### PR TITLE
fix(ci): cache fil proof parameter files

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

We've encountered quite a few CI timeouts due to slow downloads of fil proof parameter files(The slowness issue has been tracked by https://github.com/ChainSafe/forest/issues/3826). This PR tries to use build cache to avoid downloading these files to improve CI stability and reduce timecost.

Changes introduced in this pull request:

- cache proof param files in the jobs that need them.
- set `FIL_PROOFS_PARAMETER_CACHE` to `/var/tmp/filecoin-proof-parameters` to share the folder between forest and lotus
- ~~change devnet docker-compose config to mount local folder instead of volume for proof params~~
- remove `update-schedules.dff` and set `LOTUS_WATERMELON_HEIGHT=10` instead, see https://github.com/filecoin-project/lotus/blob/8f73f157933435f5020d7b8f23bee9e4ab71cb1c/build/params_2k.go#L108
- print lotus versions in `docker-compose.yml`
- upgrade lotus image tag to `2024-01-23-90ff9b2`(generated by https://github.com/ChainSafe/forest/actions/runs/7622411436)

To verify the change, search `paramfetch` in CI job logs, below lines should be gone.

```
# forest
2024-01-22T15:15:01.863850Z  INFO forest_filecoin::utils::proofs_api::paramfetch: Fetching param file "/home/runner/.local/share/forest/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-8-0-0377ded656c6f524f1618760bffe4e0a1c51d5a70c4509eedae8a27555733edc.vk" from https://proofs.filecoin.io/ipfs/
2024-01-22T15:15:01.863873Z  INFO forest_filecoin::utils::proofs_api::paramfetch: Fetching param file "/home/runner/.local/share/forest/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-8-0-559e581f022bb4e4ec6e719e563bf0e026ad6de42e56c18714a2c692b1b88d7e.vk" from https://proofs.filecoin.io/ipfs/
...

# lotus
+ lotus fetch-params 2048
...
2024-01-22T07:07:15.572Z	INFO	paramfetch	go-paramfetch@v0.0.4/paramfetch.go:262	GET https://proofs.filecoin.io/ipfs/QmYM6Hg7mjmvA3ZHTsqkss1fkdyDju5dDmLiBZGJ5pz9y9
  
   0 B / 625.41 MiB    0.00%
   992.50 KiB / 625.41 MiB    0.15% 4.[83](https://github.com/ChainSafe/forest/actions/runs/7607045683/job/20715277855?pr=3850#step:6:89) MiB/s 02m08s
   992.50 KiB / 625.41 MiB    0.15% 2.42 MiB/s 02m08s
   992.50 KiB / 625.41 MiB    0.15% 1.61 MiB/s 02m08s
   992.50 KiB / 625.41 MiB    0.15% 1.21 MiB/s 02m08s
...
```

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: scripts/devnet/docker-compose.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ services:
   lotus_init:
     image: ${LOTUS_IMAGE}
     volumes:
-      - filecoin-proofs:${FILECOIN_PROOFS_DIR}
```
Comment (LesnyRumcajs) [inline scripts/devnet/docker-compose.yml:16]: I'd rather we keep the host clean, without any intermediary artefacts generated by running the devnet. We could achieve this with, e.g., populating the volume from a container or putting the filecoin proofs in the Docker images used within the devnet. 

What do you think?

Comment (hanabi1224) [reply scripts/devnet/docker-compose.yml:16]: @LesnyRumcajs Changed to keep the `filecoin-proofs` volume and manipulate CI cache with extra steps

**Review by LesnyRumcajs — APPROVED** [review]
I didn't know about these Lotus env variables, neat.

File: scripts/devnet/ci_setup.sh [hunk 2]
```diff
@@ -15,6 +15,9 @@ export FOREST_DOCKERFILE_OVERRIDE=scripts/devnet/forest_ci.dockerfile
 # This should not be needed in GH. It is useful for running locally.
 docker compose down --remove-orphans
 docker compose rm -f
+# Cleanup data volumes
+docker volume rm -f devnet_lotus-data
+docker volume rm -f devnet_forest-data
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply scripts/devnet/ci_setup.sh:20]: @LesnyRumcajs I'd like to keep the proofs volume, to make it easier to run locally

**Review by aatifsyed — APPROVED** [review]

**Pull request merged**